instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure/build.gradle.kts

```diff
@@ -5,6 +5,7 @@ plugins {
 base.archivesName.set("${base.archivesName.get()}-autoconfigure")
 
 dependencies {
+  implementation(project(":javaagent-bootstrap"))
```

Comment:
no, it is not. library modules can not depend on `:javaagent-bootstrap`

Reply:
What is the appropriate way to provide the resource attribute for the mdc [here?](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9647/files#diff-3a60139cc05a74217fb45ac63bb746343a070ecb9700bb3f21d21175c7d87824R51)

Reply:
I think it would be acceptable to have a compile only dependency and in the code first verify that the class is present with Class.forName before using it.

Reply:
Like this?
```java

try {
      Class.forName("io.opentelemetry.javaagent.bootstrap.ConfiguredResourceAttributesHolder");
      contextData.putAll(ConfiguredResourceAttributesHolder.getResourceAttributes());
    } catch (ClassNotFoundException ignore) {      
    }
```

Reply:
typically you'd do the class presence check only once in class initializer and set a boolean based on that and use that boolean everywhere else

Reply:
@laurit how does look?